app/javascript/angular/code/services/_draw_session.js

```diff
@@ -38,14 +39,12 @@ export const drawSession = (sensors, map, heat, note, empty) => {
           suffix
         );
 
-        drawnObjects.markers.push(marker);
+        drawnObjects.push(marker);
         points.push(measurement);
       });
 
-      (session.notes || []).forEach(function(noteItem, idx) {
-        drawnObjects.noteDrawings.push(note.drawNote(noteItem, idx));
-      });
-      drawnObjects.lines.push(map.drawLine(points));
+      drawnObjects = drawnObjects.concat(drawNotes(session.notes || [], map));
```

Comment:
I tried passing the callback instead of the whole map, but map uses this.something in too many places. Maybe in the next refactor :)